driver-core/src/main/com/mongodb/internal/operation/AsyncOperationHelper.java

```diff
@@ -294,7 +299,11 @@ static <R> AsyncCallbackSupplier<R> decorateReadWithRetriesAsync(final RetryStat
 
     static <R> AsyncCallbackSupplier<R> decorateWriteWithRetriesAsync(final RetryState retryState, final OperationContext operationContext,
             final AsyncCallbackSupplier<R> asyncWriteFunction) {
-        return new RetryingAsyncCallbackSupplier<>(retryState, CommandOperationHelper::chooseRetryableWriteException,
+        BiFunction<Throwable, Throwable, Throwable> onAttemptFailure =
```

Comment:
[Optional] We could use `BinaryOperator<Throwable> onAttemptFailure` here. This modification would also necessitate changes to the constructor of `RetryingAsyncCallbackSupplier.`

P.S. The existing code prior to the changes in this PR should function without conflicts if we make this change in `RetryingAsyncCallbackSupplier.`

Reply:
At first, I made a similar change in `master`: https://github.com/mongodb/mongo-java-driver/pull/1374. Then I [made changes as you proposed](https://github.com/mongodb/mongo-java-driver/pull/1367/commits/e4ffab42ff04e943bd2b9fcdf2689ec3d499b7ad) in the PR at hand.